plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveUtil.java

```diff
@@ -374,8 +376,8 @@ public static String getInputFormatName(Properties schema)
 
     public static long parseHiveDate(String value)
     {
-        long millis = HIVE_DATE_PARSER.parseMillis(value);
-        return TimeUnit.MILLISECONDS.toDays(millis);
+        LocalDate date = HIVE_DATE_PARSER.parseLocalDate(value);
+        return Days.daysBetween(EPOCH_DAY, date).getDays();
```

Comment:
Maybe parse LDT and validate it's a whole round date?
e.g. accept `2022-04-08 00:00:00` but reject `2022-04-08 12:34:56`